packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCombobox, useMultipleSelection } from 'downshift';
+import { useState } from 'react';
+
+import { useStyles2 } from '../../themes';
+import { Checkbox } from '../Forms/Checkbox';
+import { Portal } from '../Portal/Portal';
+
+import { ComboboxOption, ComboboxBaseProps, AutoSizeConditionals, itemToString } from './Combobox';
+import { OptionListItem } from './OptionListItem';
+import { ValuePill } from './ValuePill';
+import { getMultiComboboxStyles } from './getMultiComboboxStyles';
+
+interface MultiComboboxBaseProps<T extends string | number> extends Omit<ComboboxBaseProps<T>, 'value' | 'onChange'> {
```

Comment:
Since Combobox is basically just an Input, we can't just pass it on, as we need custom styling surrounding it. When we start to implement some of the other functionality, I hope we are able to reuse some of the stuff in the Downshift hook, so then it starts to make sense. I argue that they should behave in a similar manner, hence we need the same props (width=auto, maxWidth, custom value support etc.)

Reply:
The underlying implementation should have no effect on the props. What we could do is to create a `CommonProps` interface that both single and multi extends. They both need to support auto sizing and custom value creation, along with some other stuff as some basic functionality props, such as `isLoading`, `placeholder`, `disabled` and `invalid`.

The only real difference between the two is that you can set multiple values and select multiple values with one.

Reply:
Not sure I exactly follow but it's not a big deal either way. My argument in general, if taken to a very simplified example, is that even though human and a clock may have 2 hands it does not represent the same concept so should probably not be reused even though they sound/look similar. Here I assume there is enough commonality that reusing some props makes sense. Personally, having a smaller subset of `CommonProps` probably sounds better as I assume it's not a clear subset/superset relationship, for example the `isClearable` probably makes less sense in the multicombox.

Reply:
Good point, I think we can iterate on this. Do you want me to update the types, or are you fine with approving?